src/Backend/Modules/Pages/Domain/ModuleExtra/ModuleExtraRepository.php

```diff
@@ -127,4 +128,31 @@ public function getWidgetId(string $module, string $action): ?int
 
         return $moduleExtra->getId();
     }
+
+    public function updateWidgetDataByModuleAndSequence(string $module, string $sequence, $data): void
+    {
+        $moduleExtra = $this->findOneBy(
+            [
+                'module' => $module,
+                'type' => 'widget',
+                'sequence' => $sequence,
+            ]
+        );
+
+        if (!$moduleExtra instanceof ModuleExtra) {
+            throw new RuntimeException('Widget not found');
```

Comment:
Maybe it might be good to create a `ModuleExtraNotFoundException` instead?

Reply:
Changed in f3759423